lms/djangoapps/instructor/views/api.py

```diff
@@ -726,6 +728,343 @@ def extract_user_info(user):
+    """
+    Takes a query and bundles it up into a query object
+    """
+    blocks = query_id.split("/")
+    if len(blocks) != 2:
+        return
+    else:
+        block_type, block_id = blocks
+    query_id = course_id.make_usage_key(block_type, block_id)
+    if query_filtering is None:
+        return None
+    query_filtering = query_filtering.lower().strip()
+
+    query_type = query_type.lower().strip()
+    if query_type == "section":
+        query_type = QUERY_TYPE.SECTION
+        if query_filtering == SECTION_FILTERS.OPENED.lower():
+            query_filtering = SECTION_FILTERS.OPENED
+        elif query_filtering == SECTION_FILTERS.NOT_OPENED.lower():
+            query_filtering = SECTION_FILTERS.NOT_OPENED
+        elif query_filtering == SECTION_FILTERS.COMPLETED.lower():
+            query_filtering = SECTION_FILTERS.COMPLETED
+    else:
+        query_type = QUERY_TYPE.PROBLEM
+        if query_filtering == PROBLEM_FILTERS.OPENED.lower():
+            query_filtering = PROBLEM_FILTERS.OPENED
+        elif query_filtering == PROBLEM_FILTERS.NOT_OPENED.lower():
+            query_filtering = PROBLEM_FILTERS.NOT_OPENED
+        elif query_filtering == PROBLEM_FILTERS.COMPLETED.lower():
+            query_filtering = PROBLEM_FILTERS.COMPLETED
+        elif query_filtering == PROBLEM_FILTERS.NOT_COMPLETED.lower():
+            query_filtering = PROBLEM_FILTERS.NOT_COMPLETED
+        elif query_filtering == "score":
+            query_filtering = PROBLEM_FILTERS.SCORE
+        elif query_filtering == "number of peer responses graded":
+            query_filtering = PROBLEM_FILTERS.NUMBER_PEER_GRADED
+    return Query(query_type, query_incl, query_id, query_filtering, entity_name)
+
+
+@ensure_csrf_cookie
+@cache_control(no_cache=True, no_store=True, must_revalidate=True)
+@require_level('instructor')
+def delete_bulk_temp_query(request, course_id):  # pylint: disable=unused-argument
+    """
+    Deletes a temporary query that the user has entered along with the corresponding students
+    """
+    existing = request.GET.get('existing')
+    if existing is not None and existing != "":
+        existing_queries = existing.split(',')
+    else:
+        existing_queries = []
+    cleaned_queries = [query.strip() for query in existing_queries]
+    if len(cleaned_queries) > 0:
+        data_access.delete_bulk_temporary_query(cleaned_queries)
+    response_payload = {
+        'success': True
+    }
+    return JsonResponse(response_payload)
+
+
+@ensure_csrf_cookie
+@cache_control(no_cache=True, no_store=True, must_revalidate=True)
+@require_level('instructor')
+def delete_temp_query(_request, course_id, query_to_delete):  # pylint: disable=unused-argument
+    """
+    Deletes a temporary query that the user has entered along with the corresponding students
+    """
+    data_access.delete_temporary_query(query_to_delete)
+    response_payload = {
+        'success': True
```

Comment:
I prefer to add trailing commas to all dicts/lists, as it makes the `diff` cleaner, in the event that another value is added later.